build_tools/ci/matmul_ukernel_vs_codegen_benchmark/run_test.sh

```diff
@@ -0,0 +1,253 @@
+  mkdir -p ${dir}
+  if [ ! -d "${dir}" ]; then
+    echo "Failed to locate or construct directory '${dir}'."
+    exit 1
+  fi
+done
+
+IREE_INSTALL_DIR=`realpath "$2"`
+if [ ! -d "${IREE_INSTALL_DIR}" ]; then
+  echo "IREE_INSTALL_DIR must be a directory, '${IREE_INSTALL_DIR}' is not."
+  exit 1
+fi
+
+# Search for iree-compile in the user provide directory.
+IREE_COMPILE_EXE=""
+for dir in "${IREE_INSTALL_DIR}" "${IREE_INSTALL_DIR}/bin" "${IREE_INSTALL_DIR}/tools"; do
+  if [ -f "${dir}/iree-compile" ]; then
+    IREE_COMPILE_EXE="${dir}/iree-compile"
+  fi
+done
+
+if [ -z "${IREE_COMPILE_EXE}" ]; then
+  echo "No 'iree-compile' found in any of the following directories: " \
+       "'${IREE_INSTALL_DIR}', '${IREE_INSTALL_DIR}/bin', '${IREE_INSTALL_DIR}/tools'."
+  exit 1
+fi
+
+
+# Parameter 3) <mlir-aie-install-dir>
+if [ -z "${3-}" ]; then
+  MLIR_AIE_INSTALL=`realpath .venv/lib/python3.10/site-packages/mlir_aie`
+else
+  MLIR_AIE_INSTALL=`realpath "$3"`
+fi
+if [ ! -d "${MLIR_AIE_INSTALL}" ]; then
+  echo "No directory '${MLIR_AIE_INSTALL}' (argument 3) found."
+  exit 1
+fi
+
+# Parameter 4) <peano-install-dir>
+if [ -z "${4-}" ]; then
+  PEANO=/opt/llvm-aie
+else
+  PEANO=`realpath "$4"`
+fi
+if [ ! -d "${PEANO}" ]; then
+  echo "No directory '${PEANO}' (argument 4) found."
+  exit 1
+fi
+
+# Parameter 5) <xrt-dir>
+if [ -z "${5-}" ]; then
+  XRT_DIR=/opt/xilinx/xrt
+else
+  XRT_DIR=`realpath "$5"`
+fi
+if [ ! -d "${XRT_DIR}" ]; then
+  echo "No directory '${XRT_DIR}' (argument 5) found."
+  exit 1
+fi
+
+# Parameter 6) <vitis-install-dir>
+if [ -z "${6-}" ]; then
+  # An alternate to a full vitis install, will work
+  # here but not for a full build of mlir-aie
+  # https://riallto.ai/install-riallto.html#install-riallto
+  # VITIS=/opt/Riallto/Vitis/2023.1
+  VITIS=/opt/Xilinx/Vitis/2023.2
+else
+  VITIS=`realpath "$6"`
+fi
+if [ ! -d "${VITIS}" ]; then
+  echo "No directory '${VITIS}' (argument 6) found."
+  exit 1
+fi
+
+THIS_DIR="$(cd $(dirname $0) && pwd)"
+ROOT_DIR="$(cd $THIS_DIR/../.. && pwd)"
+
+
+source $XRT_DIR/setup.sh
+export XRT_HACK_UNSECURE_LOADING_XCLBIN=1
+
+M=1024
+K=4096
+N=1024
+
+
+# Create a file ${OUTPUT_DIR}/generated_linalg_matmul.mlir
+# by running 'generate_linalg_matmul.py with arguments M, K, and N.
+GENERATE_SCRIPT="${THIS_DIR}/generate_linalg_matmul.py"
+SOURCE_MLIR_FILE="${OUTPUT_DIR}/generated_linalg_matmul.mlir"
+if [ ! -f "${GENERATE_SCRIPT}" ]; then
+  echo "generate_linalg_matmul.py not found: ${GENERATE_SCRIPT}"
+  exit 1
+fi
+GENERATE_COMMAND="python ${GENERATE_SCRIPT} M=${M} K=${K} N=${N} > ${SOURCE_MLIR_FILE}"
+echo "Executing command: $GENERATE_COMMAND"
+eval $GENERATE_COMMAND
+
+THIS="$(cd $(dirname $0) && pwd)"
+
+BASE_COMPILATION_FLAGS="-iree-hal-target-backends=amd-aie \
+-iree-amd-aie-peano-install-dir=${PEANO} \
+-iree-amd-aie-mlir-aie-install-dir=${MLIR_AIE_INSTALL} \
+-iree-amd-aie-vitis-install-dir=${VITIS} \
+-iree-amd-aie-show-invoked-commands"
```

Comment:
This is missing the path to IREE:

```bash
-iree-amd-aie-install-dir=$IREE_INSTALL_DIR
```

Reply:
Ok thanks. This is a recent flag addition afaik. Maybe I should ask/push for this PR to be landed so it doesn't go stale..

Reply:
Yes, very recent (it just started failing for me). Having this test in CI wouldn't be entirely out of the question, I don't think. It's a good reference for people trying to do some performance analysis (like myself).